internal/api/observability/conditions.go

```diff
@@ -51,3 +53,39 @@ func SetCondition(conditions *[]metav1.Condition, newCond metav1.Condition) bool
 	*conditions = append(*conditions, newCond)
 	return true
 }
+
+// PruneConditions keeps only those conditions that match an existing spec name
+func PruneConditions(conditions *[]metav1.Condition, context internalcontext.ForwarderContext) {
```

Comment:
The context isnt being used and can be removed... or  make it the only arg and iterate through them here.  alternatively:

We are already sending the right set of conditions for a given spec.  We could introduce an interface that our internal/api/obs types implement and condense this and make it generic:

```
type NameList interface {
    Names() []string
}
func (outputs Outputs) Names []string{
}
func (inputs Inputs) Names []string{
}

func PruneConditions(conditions *[]metav1.Condition, names NameList) {
	keepers := []metav1.Condition{}
	for _, condition := range *conditions {
		for _, name := range names {
			if strings.HasSuffix(condition.Type, name) {
				keepers = append(keepers, condition)
			}
		}
        }
    *conditions = keepers
}

. . .

internalobs.PruneConditions(&forwarder.Status.Inputs, forwarder.Spec.Inputs)
internalobs.PruneConditions(&forwarder.Status.Outputs, forwarder.Spec.Outputs)
...
```

Reply:
Refactored